### PR TITLE
WT-9710 Re-enable tcmalloc usage in evergreen

### PR DESCRIPTION
**Pull request by mickgraham-mongodb**
Re-enabled tcmalloc usage in evergreen. Updated linking of tcmalloc from at the shared object level to the executable level which should prevent running multiple allocators which is known to introduce issues.

### CODE REVIEW

File: test/cppsuite/CMakeLists.txt [hunk 1]
```diff
@@ -55,11 +55,6 @@ create_test_executable(csuite_style_example_test
     CXX
 )
 
-if(ENABLE_TCMALLOC)
```
Comment (quchenhao) [inline test/cppsuite/CMakeLists.txt:58]: Why we remove this from here? If we are linking at the executable level, should we keep this?

File: test/unittest/CMakeLists.txt [hunk 2]
```diff
@@ -37,9 +37,6 @@ target_compile_options(
 )
 
 target_link_libraries(unittests Catch2::Catch2)
-if(ENABLE_TCMALLOC)
```
Comment (quchenhao) [inline test/unittest/CMakeLists.txt:40]: Same, we we are removing it here?

File: test/cppsuite/CMakeLists.txt (hunk 1 above)
Comment (mickgraham-mongodb) [reply test/cppsuite/CMakeLists.txt:58]: create_test_executable is setting the option so we don't need it twice.

File: test/unittest/CMakeLists.txt (hunk 2 above)
Comment (mickgraham-mongodb) [reply test/unittest/CMakeLists.txt:40]: create_test_executable is setting the option so we don't need it twice.

**Review by quchenhao — APPROVED** [review]
lgtm

File: test/evergreen.yml [hunk 3]
```diff
@@ -4080,6 +4081,8 @@ buildvariants:
     - name: ".stress-test-sanitizer-4"
     - name: format-abort-recovery-stress-test
 
+# When running the Python tests on this variant tcmalloc must be preloaded otherwise the wiredtiger library
+# fails to load and resolve its dependency..
```
Comment (korteland) [inline test/evergreen.yml:4085]: Nitpick, double full stop.

**Review by korteland — APPROVED** [review]
LGTM! Thanks for fixing this.

File: test/evergreen.yml (hunk 3 above)
Comment (mickgraham-mongodb) [reply test/evergreen.yml]: Fixed

**Pull request merged**